syncapi/sync/notifier.go

```diff
@@ -37,8 +37,8 @@ type Notifier struct {
 	streamLock *sync.Mutex
 	// The latest sync position
 	currPos types.StreamingToken
-	// A map of user_id => UserStream which can be used to wake a given user's /sync request.
-	userStreams map[string]*UserStream
+	// A map of user_id => device_id => UserStream which can be used to wake a given user's /sync request.
+	userDeviceStreams map[string]map[string]*UserDeviceStream
```

Comment:
I'd be tempted to mux user ID / device ID together given the restricted grammar they use, it'll make it easier to work with than nested maps.

Reply:
It's a map like this so that we can wake up all user devices easily by just looping over the second map.